socorro/external/boto/connection_context.py

```diff
@@ -157,17 +221,20 @@ def fetch(self, id, name_of_thing):
         conn = self._connect()
         bucket = self._get_bucket(conn, self.config.bucket_name)
 
-        key = self.build_key(self.config.prefix, name_of_thing, id)
-        key_object = bucket.get_key(key)
-        if key_object is None:
-            raise KeyNotFound(
-                '%s (bucket=%r key=%r) not found, no value returned' % (
-                    id,
-                    self.config.bucket_name,
-                    key,
-                )
+        all_keys = self.build_keys(self.config.prefix, name_of_thing, id)
+        for key in all_keys:
+            key_object = bucket.get_key(key)
+            if key_object is not None:
+                return key_object.get_contents_as_string()
+
+        # None of the keys worked, so raise an error
+        raise KeyNotFound(
+            '%s (bucket=%r key=%r) not found, no value returned' % (
+                id,
+                self.config.bucket_name,
+                all_keys,
             )
-        return key_object.get_contents_as_string()
+        )
```

Comment:
I think I got these changes right, but double-checking them and thinking about scenarios where this fails would be helpful.